smartsim/entity/files.py

```diff
@@ -130,3 +130,95 @@ def _check_path(self, file_path):
         if path.isdir(full_path):
             return full_path
         raise SSConfigError(f"File or Directory {file_path} not found")
+
+
+class TaggedFilesHierarchy:
+    """TaggedFilesHierarchy maintains a list of files and a list of child
+    TaggedFilesHierarchies. The TaggedFilesHierarchy class is ment to be easily
+    converted into a directory file structure each with each dir containing a
+    copy of the specifed (possilby tagged) files specifed at each level of the
+    hierarchy via a simply depth first search over the hierarchy.
+    """
+
+    def __init__(self, base=""):
+        """Initialize a TaggedFilesHierarchy
+
+        :param base: pathlike string specifing the generated directory
+                     files are located
+        :type base: str, optional
+        """
+        self.base = base
+        self.files = set()
+        self.dirs = set()
+
+    @classmethod
+    def from_list_paths(cls, path_list, dir_contents_to_base=False):
+        """Given a list of absolute paths to files and dirs, create and return
+        a TaggedFilesHierarchy instance to representing the file hierarchy
+
+        :param path_list: list of absolute path like strings to tagged files
+                          or dirs containing tagged files
+        :type path_list: list of str
+        :param dir_contents_to_base: When a top level dir is encountered, if
+                                     this value is truthy, files in the dir are
+                                     put into the base hierarchy level.
+                                     Otherwise, a new sub level is created for
+                                     the dir
+        :type dir_contents_to_base: bool
+        :return: A built tagged file hierarchy for the given files
+        :rtype: TaggedFilesHierarchy
+        """
+        th = cls()
+        if dir_contents_to_base:
+            new_paths = []
+            for p in path_list:
+                if path.isdir(p):
+                    new_paths += [path.join(p, f) for f in os.listdir(p)]
+                else:
+                    new_paths.append(p)
+            path_list = new_paths
+        th._add_paths(path_list)
+        return th
+
+    def _add_file(self, f):
```

Comment:
I'd suggest using a meaningful name for the argument, thus `f`->`file`?

Reply:
resolved in https://github.com/CrayLabs/SmartSim/pull/119/commits/3cf5b79d4a70cf8019631dea7a0e8641a082ac92